script.js

```diff
@@ -1,12 +1,250 @@
+        episode.name.toLowerCase().includes(searchString) ||
+        episode.summary.toLowerCase().includes(searchString)
+    );
+    // now I have my filtered data, present on screen
+    makePageForEpisodes(filteredNames);
+    //update total for all number of episodes
+    updateNumberOfEpisodes(filteredNames.length, epList.length);
+  });
 }
 
 function makePageForEpisodes(episodeList) {
-  const rootElem = document.getElementById("root");
-  rootElem.textContent = `Got ${episodeList.length} episode(s)`;
+  // store html string in variable
+  let html = "";
+  // loop through episodes & append  html to variable
+  episodeList.forEach((episode) => {
+    html += `<div class= "episodeGrid"> <h2>${episode.name} 
+    - Season ${seasonNumbers(episode.season)}  
+    Episode ${seasonNumbers(episode.number)} 
+    </h2>  ${
+      episode.image && `<img src = "${episode.image.medium} " alt = " ">`
+    }
+    ${episode.summary}</div>`;
+  });
+  // insert my search results 'html' into the user interface.
+  rootElem.innerHTML = html;
+}
+
+// adds the episode and season
+function seasonNumbers(seasonNum) {
+  return seasonNum.toString().padStart(2, 0);
+}
+
+//adding data to the drop down menu
+function watchShow(watch) {
+  let tvEpisode = `<option value="0" >See all episodes</option>`;
+
+  // displays Episode name and Season number
+  watch.forEach((episode) => {
+    tvEpisode += `<option value="${episode.id}">
+    Season ${seasonNumbers(episode.season).toString().padStart(2, 0)}  
+    Episode ${seasonNumbers(episode.number).toString().padStart(2, 0)}
+    ${episode.name}</option>`;
+  });
+
+  selection.insertAdjacentHTML("afterbegin", tvEpisode);
+}
+
+// this function displays the list of shows in the first drop down box starting from see all shows
+function listShows(shows) {
+  let tvShow = `<option value="0" >See all shows</option>`;
+
+  // displays series name on the first drop down box
+  shows
+    .sort(function (a, b) {
+      a.name.toLowerCase();
+      b.name.toLowerCase();
+      if (a.name.toLowerCase() < b.name.toLowerCase()) {
+        return -1;
+      }
+      if (a.name.toLowerCase() > b.name.toLowerCase()) {
+        return 1;
+      }
+      return 0;
+    })
+    .forEach((series) => {
+      tvShow += `<option value="${series.id}"> 
+    ${series.name}</option>`;
+    });
+
+  showsSelect.insertAdjacentHTML("afterbegin", tvShow);
+}
+
+//when the user clicks on the drop down menu, episode data can be selected
+function dropDown(shows) {
+  selection.addEventListener("change", (m) => {
+    const searchId = +m.target.value;
+    let filteredList = [];
+    if (searchId === 0) {
+      filteredList = shows;
+    } else {
+      filteredList = shows.filter((popcorn) => {
+        return popcorn.id === searchId;
+      });
+    }
+    rootElem.innerHTML = "";
+    makePageForEpisodes(filteredList);
+  });
+}
+
+// when the user clicks on the first drop down menu, new shows can be selected
+function dropDown2() {
```

Comment:
I would call these functions something more helpful rather than dropDown and dropDown2